### PR TITLE
fix: pass ip version to blocklist check to ensure ipv6 accepted

### PR DESCRIPTION
**Pull request by Jafferwaffer**
Issue: https://brandwatch.atlassian.net/browse/AS-454

``` JS
> const blockList = new net.BlockList();
> blockList.addSubnet('10.0.0.0', 8);
> blockList.check('::ffff:10.220.73.9')
false
> blockList.check('::ffff:10.220.73.9', 'ipv6')
true
> blockList.check('::ffff:10.220.73.9', 'ipv4')
false
> net.isIPv6('::ffff:10.220.73.9');
true
> net.isIPv4('::ffff:10.220.73.9');
false
```

### CODE REVIEW

**Comment by Jafferwaffer** [comment]
IPs from kubelet (which does healthz checks) are ipv6 in the stage apps cluster & likely the other clusters too.

**Comment by Jafferwaffer** [comment]
See API here https://nodejs.org/docs/latest-v20.x/api/net.html#blocklistcheckaddress-type

File: src/index.js [hunk 1]
```diff
@@ -40,7 +40,9 @@ const getMiddleware = () => (req, res, next) => {
                        || req.socket.remoteAddress || req.connection.socket.remoteAddress;
 
   // Only expect private IPs from this range
-  if (!blocklist.check(requestingIP)) {
+  const ipVerion = net.isIPv6(requestingIP) ? 'ipv6' : 'ipv4';
```
Comment (SeaOfTea) [inline src/index.js:43]:
```suggestion
  const ipVersion = net.isIPv6(requestingIP) ? 'ipv6' : 'ipv4';
```

File: src/index.js [hunk 2]
```diff
@@ -40,7 +40,9 @@ const getMiddleware = () => (req, res, next) => {
...
+
+  if (!blocklist.check(requestingIP, ipVerion)) {
```
Comment (SeaOfTea) [inline src/index.js:45]:
```suggestion
  if (!blocklist.check(requestingIP, ipVersion)) {
```

**Review by SeaOfTea — COMMENTED** [review]
:eliot:

**Review by SeaOfTea — APPROVED** [review]

**Pull request merged**